frontend/packages/dev-console/src/components/edit-application/edit-application-utils.ts

```diff
@@ -0,0 +1,150 @@
+import * as _ from 'lodash';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { BuildStrategyType } from '@console/internal/components/build';
+import { Resources } from '../import/import-types';
+
+export const getResourcesType = (resource: K8sResourceKind): string => {
+  switch (resource.kind) {
+    case 'DeploymentConfig':
+      return Resources.OpenShift;
+    case 'Deployment':
+      return Resources.Kubernetes;
+    default:
+      return Resources.KnativeService;
```

Comment:
Very generous to knative :P
Probably best to be resilient and do an explicit comparison and return null if not found or throw an error if it should never be the case.